src/praisonai-agents/praisonaiagents/memory/adapters/perseus_vault_adapter.py

```
@@ -0,0 +1,282 @@
+        self._env = {**os.environ, **(env or {})}
+        self._lock = threading.Lock()
+        self._id = 0
+        self._proc: Optional[subprocess.Popen] = None
+        self._startup_timeout = startup_timeout
+        self._start()
+
+    def _start(self) -> None:
+        cmd = [self._binary, "serve", "--db", self._db_path]
+        if self._encryption_key:
+            cmd += ["--encryption-key", self._encryption_key]
+        self._proc = subprocess.Popen(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)